soroban-env-host/src/native_contract/base_types.rs

```diff
@@ -4,9 +4,52 @@ use core::cmp::Ordering;
 use soroban_env_common::xdr::{AccountId, ScAddress};
 use soroban_env_common::{
     AddressObject, BytesObject, Compare, ConversionError, Env, EnvBase, MapObject, RawVal,
-    TryFromVal, VecObject,
+    Symbol as SymbolVal, SymbolSmall, TryFromVal, VecObject,
 };
 
+#[derive(Clone)]
+pub struct Symbol {
+    val: SymbolVal,
+}
+
+impl TryFromVal<Host, SymbolVal> for Symbol {
+    type Error = HostError;
+
+    fn try_from_val(_env: &Host, val: &SymbolVal) -> Result<Self, Self::Error> {
+        Ok(Symbol { val: *val })
+    }
+}
+
+impl TryFromVal<Host, RawVal> for Symbol {
+    type Error = HostError;
+
+    fn try_from_val(env: &Host, val: &RawVal) -> Result<Self, Self::Error> {
+        let val = *val;
+        let obj: SymbolVal = val.try_into()?;
+        Symbol::try_from_val(env, &obj)
+    }
+}
+
+impl TryFromVal<Host, Symbol> for RawVal {
+    type Error = ConversionError;
+
+    fn try_from_val(_env: &Host, v: &Symbol) -> Result<Self, Self::Error> {
+        Ok(v.val.into())
+    }
+}
+
+impl Symbol {
+    pub const fn short(s: &str) -> Self {
```

Comment:
I'm not sure this is safe as it can be called at runtime. It doesn't seem like you generally need this wrapper type for the host, do you? The main value of introducing this in SDK was to avoid unwraps and generalize short and long symbols, both of which aren't necessary for the host.

Reply:
Good point. I removed the Symbol base_type.